### DISCUSSION TITLE
Msgpack Content

### DISCUSSION THREAD

**Discussion by 93578237** [Neoteroi/BlackSheep#204] [Ideas]
Feature: `MsgpackContent`
Hi, thx for your work!
Do you mind supporting `msgpack` as a separate Content class.

**Comment by 93578237** [comment]
I started implementing it, but official lib [msgpack](https://github.com/msgpack/msgpack-python) lacks of builtin support of custom data types. [msgpec](https://github.com/jcrist/msgspec) and [ormsgpack](https://github.com/aviramha/ormsgpack) can do it

**Reply by RobertoPrevato** [reply]
Hi!
Thanks for the idea, I didn't know MessagePack specification and it looks interesting. At first sight, maybe a dedicated package would be a good fit for this case? I think most applications will use JSON anyway, and right now BlackSheep doesn't even have built-in support for XML handling out of the box.

The benefit I see of of having a dedicated package (like I did for BlackSheep-SQLAlchemy) is that it's easier to maintain since it doesn't add extra optional dependency to the core library.

**Reply by RobertoPrevato** [reply]
If we keep it in a dedicated library, I could create a repository in the same organization, and prepare the same GitHub Workflows for CI/CD, what do you think?
